### PR TITLE
Implementing notify for Incoming Calls

### LINKED ISSUE #128: is there anyway to get incoming call event on iOS?
on iOS I want to get notify when get voip notification, so that I can track if the call is cancelled by the caller. Is there anyway to do it?

### PR DESCRIPTION
**Pull request by tungs0ul**
This pull request introduces the ability to notify Flutter from iOS when there is an incoming call.

closes #128

### CODE REVIEW

**Review by TatankaConCube — CHANGES_REQUESTED** [review]
the feature should be implemented for both supported platforms, for android it should work in the terminated state too

File: android/src/main/kotlin/com/connectycube/flutter/connectycube_flutter_call_kit/ConnectycubeFCMReceiver.kt [hunk 1]
```diff
@@ -84,6 +88,30 @@ class ConnectycubeFCMReceiver : BroadcastReceiver() {
             return
         }
 
+        if (data["signal_type"] == "startCall") {
```
Comment (TatankaConCube) [inline android/src/main/kotlin/com/connectycube/flutter/connectycube_flutter_call_kit/ConnectycubeFCMReceiver.kt:91]: this statement check is redundant cause we did it before in line 37

File: android/src/main/kotlin/com/connectycube/flutter/connectycube_flutter_call_kit/ConnectycubeFCMReceiver.kt [hunk 2]
```diff
@@ -84,6 +88,30 @@ class ConnectycubeFCMReceiver : BroadcastReceiver() {
...
+                intent.putExtra("userCallbackHandleName", INCOMING_IN_BACKGROUND)
+                ConnectycubeFlutterBgPerformingService.enqueueMessageProcessing(
+                    applicationContext,
+                    intent
+                )
+            }
+
+            Log.d(TAG, "[processInviteCallEvent] sendBroadcast ACTION_CALL_INCOMING $callId")
+        }
+
```
Comment (TatankaConCube) [inline android/src/main/kotlin/com/connectycube/flutter/connectycube_flutter_call_kit/ConnectycubeFCMReceiver.kt:114]: could you please create the separate method/function (like 'notifyAboutIncomingCall' or something like this) in the `ConnectycubeFlutterCallKitPlugin.kt` similar to `saveCallState` and call it here. don't forget to delete unused imports

**Comment by tungs0ul** [comment]
> @tungs0ul have you checked the new feature on your side for both platforms?

yup i have tested for both

**Comment by TatankaConCube** [comment]
@tungs0ul just now version [2.6.0](https://pub.dev/packages/connectycube_flutter_call_kit/versions/2.6.0) with your changes was published, thank you for your contribution

**Pull request merged**